version: '3'

services:
    postgres:
        restart: on-failure
        image: postgres
        environment:
            - POSTGRES_DB=telemetry
            - POSTGRES_USER=postgres
            - POSTGRES_PASSWORD=postgres
        networks:
            - main
        ports:
            - "5432:5432"
    test:
        build:
            context: .
            args:
                requirements: Pipfile.lock
        environment:
            - ENVIRONMENT=ci
            - SECRET_KEY=secret-key-example
            - DATABASE_URL=postgres://postgres:postgres@postgres:5432/postgres
            - DJANGO_SETTINGS_MODULE=server.settings
            - LOG_DIR=
            - ALLOWED_HOSTS=*
            - DJANGO_LOG_LEVEL=INFO
            - VERSION=0.0.0
            - DEBUG=
        command: >
            bash -c "pipenv run flake8 && pipenv run python manage.py test"
        volumes:
            - .:/build
        networks:
            - main
        depends_on:
            - postgres
        links:
            - postgres
networks:
    main:
